pyfixest/estimation/feglm_.py

```diff
@@ -0,0 +1,520 @@
+from abc import ABC, abstractmethod
+from typing import Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from pyfixest.errors import (
+    NonConvergenceError,
+)
+from pyfixest.estimation.demean_ import demean
+from pyfixest.estimation.feols_ import Feols, PredictionType
+from pyfixest.estimation.fepois_ import _check_for_separation
+
+# from pyfixest.estimation.fepois_ import Fepois
+from pyfixest.estimation.FormulaParser import FixestFormula
+from pyfixest.utils.dev_utils import DataFrameType
+
+
+class Feglm(Feols, ABC):
+    "Abstract base class for the estimation of a fixed-effects GLM model."
+
+    def __init__(
+        self,
+        FixestFormula: FixestFormula,
+        data: pd.DataFrame,
+        ssc_dict: dict[str, Union[str, bool]],
+        drop_singletons: bool,
+        drop_intercept: bool,
+        weights: Optional[str],
+        weights_type: Optional[str],
+        collin_tol: float,
+        fixef_tol: float,
+        lookup_demeaned_data: dict[str, pd.DataFrame],
+        tol: float,
+        maxiter: int,
+        solver: str = "np.linalg.solve",
+        store_data: bool = True,
+        copy_data: bool = True,
+        lean: bool = False,
+        sample_split_var: Optional[str] = None,
+        sample_split_value: Optional[Union[str, int]] = None,
+        separation_check: Optional[list[str]] = None,
+    ):
+        super().__init__(
+            FixestFormula=FixestFormula,
+            data=data,
+            ssc_dict=ssc_dict,
+            drop_singletons=drop_singletons,
+            drop_intercept=drop_intercept,
+            weights=weights,
+            weights_type=weights_type,
+            # tol=tol,
+            # maxiter=maxiter,
+            collin_tol=collin_tol,
```

Comment:
Why these commented lines ? :)

Reply:
It's a relict from an earlier implementation in which the Feglm class inherited from Fepois, which has these arguments. Now it inherits from Feols, which does not have them, so both arguments are set in Feglm.__init__. I've cleaned this up, coming with the next commit.